openssl/src/x509/mod.rs

```diff
@@ -78,6 +78,29 @@ impl X509StoreContextRef {
     pub fn error_depth(&self) -> u32 {
         unsafe { ffi::X509_STORE_CTX_get_error_depth(self.as_ptr()) as u32 }
     }
+
+    pub fn get_chain(&self) -> Option<&mut StackRef<X509Ref>> {
```

Comment:
OpenSSL is indeed very inconsistent about its const use; I can only think of a small handful of functions that actually take const pointers to things. I find it better to go off of the logical mutability of the operation rather than the const or non constness of the pointers. In particular, it's very rare to have a function that gets a mutable reference from a shared reference.